docs/reference/aggregations/metrics/cardinality-aggregation.asciidoc

```diff
@@ -150,10 +150,15 @@ public static void main(String[] args) {
 
 image:images/cardinality_error.png[]
 
-For all 3 thresholds, counts have been accurate up to the configured threshold
-(although not guaranteed, this is likely to be the case). Please also note that
-even with a threshold as low as 100, the error remains very low, even when
-counting millions of items.
+For all 3 thresholds, counts have been accurate up to the configured threshold.
+Although not guaranteed, this is likely to be the case.  Accuracy in practice depends
+on the dataset in question.  Because the HyperLogLog++ algorithm depends on the
+leading zeros of hashed values, the exact distributions of hashes in a dataset
+can affect how accurate the cardinality is.  In general, most datasets show
+consistently good accuracy.
```

Comment:
I wonder if this detail is too much for most users:
```
Because the HyperLogLog++ algorithm depends on the
leading zeros of hashed values, the exact distributions of hashes in a dataset
can affect how accurate the cardinality is.
```
Unless you understand how HyperLogLog++ works under the hood I don't think this sentence will mean a lot. Maybe we should separate this from the rest of the explanation:

```suggestion
For all 3 thresholds, counts have been accurate up to the configured threshold.
Although not guaranteed, this is likely to be the case.  Accuracy in practice depends
on the dataset in question.  In general, most datasets show consistently good 
accuracy. Also note that even with a threshold as low as 100, the error
remains very low, even when counting millions of items.

The HyperLogLog++ algorithm depends on the leading zeros of hashed
values, the exact distributions of hashes in a dataset can affect the 
accuracy of the cardinality.  
```

WDYT?

Reply:
:+1: that github "apply suggestion" thinger is pretty neat too :)